examples/circuit-relaying/README.md

```diff
@@ -0,0 +1,303 @@
+Swarm listening on /p2p-circuit/ip4/127.0.0.1/tcp/4003/ws/ipfs/QmfQj8YwDdy1uP2DpZBa7k38rSGPvhHiC52cdAGWBqoVpq
+Swarm listening on /ip4/127.0.0.1/tcp/4003/ws/ipfs/QmfQj8YwDdy1uP2DpZBa7k38rSGPvhHiC52cdAGWBqoVpq
+Swarm listening on /ip4/127.0.0.1/tcp/4002/ipfs/QmfQj8YwDdy1uP2DpZBa7k38rSGPvhHiC52cdAGWBqoVpq
+Swarm listening on /ip4/192.168.1.132/tcp/4002/ipfs/QmfQj8YwDdy1uP2DpZBa7k38rSGPvhHiC52cdAGWBqoVpq
+API is listening on: /ip4/127.0.0.1/tcp/5002
+Gateway (readonly) is listening on: /ip4/127.0.0.1/tcp/9090
+Daemon is ready
+```
+
+Look out for an address similar to `/ip4/127.0.0.1/tcp/4003/ws/ipfs/Qm...` note it down somewhere, and lets move on to the next step. 
+
+### Configure and run the bundled example
+
+Now that we have ipfs installed and initialized, lets set up the included example. This is a standard npm package, so the usual `npm install` should get us going. Lets `cd` into the `examples/circuit-relaying` directory and run:
+
+```
+npm install
+```
+
+After it finishes, we should be able to run the project with `npm start` and get a similar output to the bellow one:
+
+```
+npm run start
+Server running at http://localhost:1234
+```
+
+The bundled example is a simple chat app that uses another cool ipfs feature - [pubsub](https://github.com/libp2p/specs/tree/master/pubsub). Lets open up a browser and paste the above url into the address bar. We should see something similar to the following image:
+
+![](./img/img1.png)
+
+### Connect the two browser nodes to the circuit relay
+
+In order for our browser nodes to be able to send messages to each other, we need to first get them connected, but for that we need to use the relay, remember, browser nodes can't be dialed directly because of lack of socket support, so the relay is here to solve that.
+
+Enter the `/ip4/127.0.0.1/tcp/4003/ws/ipfs/...` address noted above into the `Connect to Peer` field and hit the connect button:
+
+![](./img/img3.png)
+
+After connecting to the IPFS node, we should see the peer show up under the `Peers Connected` box.
+
+![](./img/img4.png)
+
+Now lets repeat the same steps with the second tab. After that, both of our browser nodes should be connected and we can move on to the next step.
+
+### Dial the two browser nodes using a `/p2p-circuit` address
+
+Having both browsers running side by side (as shown in the first screenshot), lets get them connected to each other. Head out to the `Addresses` box in one of the tabs, copy the `/p2p-circuit` address and then paste it into the `Connect to Peer` box in the other tab. Repeat these steps on the second tab.
+
+![](./img/img5.png)
+
+Lets hit the `Connect` button on each of the tabs and we should get the two browsers connected and join the chat room.
+
+![](./img/img6.png)
+
+### Send data browser to browser.
+
+Now that we have the two browsers connected, lets try the app out. Type a few words in one of the browser windows and you should see them appear in the other as well!
+
+![](./img/img7.png)
+
+Thats it!
```

Comment:
I know this tutorial gets a little long, but it might still be helpful to call out what happened in the browser code that was important. I *think* that’s:

1. Set the right config options in `new IPFS(config)`: https://github.com/ipfs/js-ipfs/blob/feat/circuit-interop/examples/circuit-relaying/src/app.js#L31-L39

2. The `/p2p-circuit/...` addresses came from calling `ipfs.id`: https://github.com/ipfs/js-ipfs/blob/feat/circuit-interop/examples/circuit-relaying/src/app.js#L52-L59

3. We called `ipfs.swarm.connect(address)` to connect, which should be nothing special (all the magic is in the address itself!) https://github.com/ipfs/js-ipfs/blob/feat/circuit-interop/examples/circuit-relaying/src/app.js#L102-L107

4. Sending and receiving don’t actually require any addressing because we are [broadcasting to anyone listening on the room name](https://github.com/ipfs/js-ipfs/blob/feat/circuit-interop/examples/circuit-relaying/src/helpers.js#L42-L49) and, on the other side, just [listening on the room name](https://github.com/ipfs/js-ipfs/blob/feat/circuit-interop/examples/circuit-relaying/src/helpers.js#L32-L36).

Did I get that right?

Reply:
I think this is a great idea - I'll add an explanation of what the browser is doing! 

And you got it right! 💯